### PR TITLE
Basic draft for plugins UT

### PR DESCRIPTION
**Pull request by vncoelho**
This is a basic draft with UT for all plugins.

I am not expert in these UT, but maybe, even without real UT tests, this will help to ensure that the class is being initialized correctly.

### CODE REVIEW

**Comment by vncoelho** [comment]
How to get these GUID for each project? @jsolman

**Comment by vncoelho** [comment]
still with errors on some plugin due to:
`: error CS0122: 'Settings' is inaccessible due to its protection level [/opt/neo-plugins/plugins.UnitTests/StatesDumper.UnitTests/StatesDumper.UnitTests.csproj]`

**Comment by vncoelho** [comment]
Protocols are duplicated in each plugin UT, this could be improved.

**Comment by vncoelho** [comment]
Most part of the UT tests are now working, problems are still found at OnImport and LogReader.

**Comment by vncoelho** [comment]
For simplification, I thought about creating all UT in the same folder, as in NEO project.

```cs
         UT_ApplicationLogs.cs(4,32): error CS0433: The type 'Settings' exists in both 'ApplicationLogs, Version=2.9.4.0, Culture=neutral, PublicKeyToken=null' and 'ImportBlocks, Version=2.9.4.0, Culture=neutral, PublicKeyToken=null' [/opt/neo-plugins/plugins.UnitTests/plugins.UnitTests.csproj]
         UT_ImportBlocks.cs(4,32): error CS0433: The type 'Settings' exists in both 'ApplicationLogs, Version=2.9.4.0, Culture=neutral, PublicKeyToken=null' and 'ImportBlocks, Version=2.9.4.0, Culture=neutral, PublicKeyToken=null' [/opt/neo-plugins/plugins.UnitTests/plugins.UnitTests.csproj]
         UT_RpcSecurity.cs(3,32): error CS0433: The type 'Settings' exists in both 'ApplicationLogs, Version=2.9.4.0, Culture=neutral, PublicKeyToken=null' and 'ImportBlocks, Version=2.9.4.0, Culture=neutral, PublicKeyToken=null' [/opt/neo-plugins/plugins.UnitTests/plugins.UnitTests.csproj]
         UT_SimplePolicy.cs(7,32): error CS0433: The type 'Settings' exists in both 'ApplicationLogs, Version=2.9.4.0, Culture=neutral, PublicKeyToken=null' and 'ImportBlocks, Version=2.9.4.0, Culture=neutral, PublicKeyToken=null' [/opt/neo-plugins/plugins.UnitTests/plugins.UnitTests.csproj]
         UT_StatesDumper.cs(4,32): error CS0433: The type 'Settings' exists in both 'ApplicationLogs, Version=2.9.4.0, Culture=neutral, PublicKeyToken=null' and 'ImportBlocks, Version=2.9.4.0, Culture=neutral, PublicKeyToken=null' [/opt/neo-plugins/plugins.UnitTests/plugins.UnitTests.csproj]
```

What is the best way to handle this?

**Comment by igormcoelho** [comment]
Brother, I think each plugin should have its own plugin unit test, because each one is an independent project. This will also resolve the Settings problem.

File: neo-plugins.sln [hunk 1]
```diff
@@ -3,17 +3,15 @@ Microsoft Visual Studio Solution File, Format Version 12.00
...
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "RpcSecurity", "RpcSecurity\RpcSecurity.csproj", "{6800D782-8EC0-49E9-98C4-195C8F781A1F}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "RpcSecurity", "RpcSecurity\RpcSecurity.csproj", "{6800D782-8EC0-49E9-98C4-195C8F781A1F}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "StatesDumper", "StatesDumper\StatesDumper.csproj", "{86531DB1-A231-46C4-823F-BE60972F7523}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "StatesDumper", "StatesDumper\StatesDumper.csproj", "{86531DB1-A231-46C4-823F-BE60972F7523}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ImportBlocks", "ImportBlocks\ImportBlocks.csproj", "{B7A42984-57BB-4F8D-967B-23B0E841B726}"
-EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SimplePolicy.UnitTests", "SimplePolicy.UnitTests\SimplePolicy.UnitTests.csproj", "{0DA81327-DC81-4960-91C8-4C5F8B9B804C}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ImportBlocks", "ImportBlocks\ImportBlocks.csproj", "{B7A42984-57BB-4F8D-967B-23B0E841B726}"
```
Comment (erikzhang) [inline neo-plugins.sln]: Unexpected changes to .sln

Comment (vncoelho) [reply neo-plugins.sln]: I do not know how to handle this, @vncoelho, monodevelop is doing a mess on all this...aheuahuea

**Comment by erikzhang** [comment]
Please add a real unit test instead of an empty project. And create pr separately.

File: neo-plugins.sln (hunk 1 above)
Comment (erikzhang) [reply neo-plugins.sln]: Maybe you can try to use Visual Studio Code. https://code.visualstudio.com/

**Comment by vncoelho** [comment]
This was a draft, @erikzhang, but drafts can not be updated here in github, thus I changed to PR.

However, the idea is to first create the template for all projects and then we adds basic tests for each project.

But maybe it is better to add each of them separately, we need, at least, to define the folder organization that will be used.

File: neo-plugins.sln (hunk 1 above)
Comment (vncoelho) [reply neo-plugins.sln]: Thanks, I will check the Debian package.

**Comment by erikzhang** [comment]
Adding each of them separately is better.

**Comment by vncoelho** [comment]
Ok, Erik.
Let's close this.

In the first one that we add maybe we can follow this standard of merging them in a plugins.UnitTest folder.
